pkg/network/server.go

```diff
@@ -224,7 +226,9 @@ func newServerFromConstructors(config ServerConfig, chain Ledger, stSync StateSy
 	s.bSyncQueue = bqueue.New(s.stateSync, log, nil, updateBlockQueueLenMetric)
 
 	s.bFetcherQueue = bqueue.New(chain, log, nil, updateBlockQueueLenMetric)
-	s.blockFetcher = blockfetcher.New(chain, s.NeoFSBlockFetcherCfg, s.bFetcherQueue, log)
+	s.blockFetcher = blockfetcher.New(chain, s.NeoFSBlockFetcherCfg, s.bFetcherQueue, log, func() {
+		s.blockFetcherFin <- struct{}{}
```

Comment:
Just this channel here in order to avoid possible fetcher deadlock on sending (if receiver is not ready to read).

Reply:
I meant close `blockFetcherFin` channel.